### PR TITLE
Add name and version

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR adds name and version to the Cache and add a new Dataset. Plus, some folder are being renamed.

Fixes #\<issue_number>

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
:books: Documentation preview :books:: https://pytorch-lightning--18796.org.readthedocs.build/en/18796/

cc @borda

### CODE REVIEW

File: src/lightning/data/streaming/cache.py [hunk 1]
```diff
@@ -48,10 +59,20 @@ def __init__(
         super().__init__()
         if not _TORCH_GREATER_EQUAL_2_1_0:
             raise ModuleNotFoundError("PyTorch version 2.1 or higher is required to use the cache.")
+
```
Comment (awaelchli) [inline src/lightning/data/streaming/cache.py:62]: Shouldn't we cover the newly added logic in a test?

**Review by awaelchli — APPROVED** [review]

**Review by ethanwharris — APPROVED** [review]

File: src/lightning/data/streaming/cache.py (hunk 1 above)
Comment (tchaton) [reply src/lightning/data/streaming/cache.py:62]: done !

File: src/lightning/data/streaming/cache.py [hunk 2]
```diff
@@ -40,6 +49,8 @@ def __init__(
             cache_dir: The path to where the chunks will be stored.
             remote_dir: The path to a remote folder where the data are located.
                 The scheme needs to be added to the path.
+            name: The name of dataset in the cloud.
```
Comment (justusschock) [inline src/lightning/data/streaming/cache.py:52]: can you add what the default corresponds to in the docstrings? Like if I don't pass `name=...` what will happen?

**Review by justusschock — APPROVED** [review]

File: src/lightning/data/streaming/cache.py (hunk 2 above)
Comment (tchaton) [reply src/lightning/data/streaming/cache.py:52]: Will do in another PR.

**Pull request merged**